cmd/acra-key-tool/acra-key-tool.go

```diff
@@ -59,66 +67,98 @@ var keyKinds = strings.Join([]string{
 	keyZonePrivate,
 }, ", ")
 
+var destroyKeyKinds = strings.Join([]string{
+	keyTransportConnector,
+	keyTransportServer,
+	keyTransportTranslator,
+}, ", ")
+
 type commandLineParams struct {
 	KeyStoreVersion string
 	KeyDir          string
 	KeyDirPublic    string
-	ClientID        string
-	ZoneID          string
-	KeyKind         string
+
+	ClientID string
+	ZoneID   string
+
+	ReadKeyKind    string
+	DestroyKeyKind string
 }
 
 var params commandLineParams
 
+func setDefaultParameterValues() {
+	if params.KeyStoreVersion == "" {
+		if filesystemV2.IsKeyDirectory(params.KeyDir) {
+			params.KeyStoreVersion = "v2"
+		} else {
+			params.KeyStoreVersion = "v1"
+		}
+	}
+	if params.KeyDirPublic == "" {
+		params.KeyDirPublic = params.KeyDir
+	}
+}
+
+func checkParameterConsistency() {
+	if params.ClientID != "" && params.ZoneID != "" {
+		log.Fatal("--client_id and --zone_id cannot be used simultaneously")
+	}
+	if params.ReadKeyKind != "" && params.DestroyKeyKind != "" {
+		log.Fatal("--read_key and --destroy_key cannot be used simultaneously")
+	}
+}
+
 func main() {
 	flag.StringVar(&params.KeyStoreVersion, "keystore", "", "force key store format: v1 (current), v2 (new)")
 	flag.StringVar(&params.KeyDir, "keys_dir", defaultKeyDir, "path to key directory")
 	flag.StringVar(&params.KeyDirPublic, "keys_dir_public", "", "path to key directory for public keys")
 	flag.StringVar(&params.ClientID, "client_id", "", "client ID for which to retrieve key")
 	flag.StringVar(&params.ZoneID, "zone_id", "", "zone ID for which to retrieve key")
-	flag.StringVar(&params.KeyKind, "key", "", "key kind to read, one of: "+keyKinds)
+	flag.StringVar(&params.ReadKeyKind, "read_key", "", "key kind to read, one of: "+readKeyKinds)
+	flag.StringVar(&params.DestroyKeyKind, "destroy_key", "", "key kind to destroy, one of: "+destroyKeyKinds)
 	err := cmd.Parse(defaultConfigPath, serviceName)
 	if err != nil {
 		log.WithError(err).
 			WithField(logging.FieldKeyEventCode, logging.EventCodeErrorCantReadServiceConfig).
 			Fatal("Cannot parse arguments")
 	}
 
-	if params.ClientID != "" && params.ZoneID != "" {
-		log.Fatal("--client_id and --zone_id cannot be used simultaneously")
-	}
-	if params.KeyStoreVersion == "" {
-		if filesystemV2.IsKeyDirectory(params.KeyDir) {
-			params.KeyStoreVersion = "v2"
-		} else {
-			params.KeyStoreVersion = "v1"
+	setDefaultParameterValues()
+	checkParameterConsistency()
+
+	if params.ReadKeyKind != "" {
+		keyBytes := readKeyBytes()
+		_, err = os.Stdout.Write(keyBytes)
```

Comment:
can we zeroize key after printing to not leave it in memory after process exit?

Reply:
> can we zeroize key after printing to not leave it in memory after process exit?

Sure. That's a good precaution. Done.